### ISSUE TITLE
Test datasets included in conda package greatly inflate installed package size

### ISSUE THREAD

**Issue by mx-moth** [conda-forge/esmpy-feedstock#78]
### Solution to issue cannot be found in the documentation.

- [X] I checked the documentation.

### Issue

When packaging esmpy for conda, the pytest tests are run before the package is created. This results in some unit test data files being downloaded and included in the final package. This inflates the compressed package size from ~3MB to ~45MB, and the installed package size from ~4MB to ~161MB.

The tests are run as part of `build.sh`:
https://github.com/conda-forge/esmpy-feedstock/blob/a323f1f4ecf6265346d9cb25d7083b068ad86121/recipe/build.sh#L13-L16

`$CONDA_BUILD_CROSS_COMPILATION` is not set, so `make test_serial` is run. This runs some tests as defined in `src/addon/esmpy/Makefile`. These tests download the test datasets in to the `lib/pythonX.Y/site-packages/esmpy/data` directory, causing them to be included in the final conda package.

Most of these tests are later run again as part of the `test:` block in the recipe. These tests are run _after_ the package has been created, in a clean environment, specifically to avoid polluting the final conda package with test artefacts like these.

If these testing datasets are not an essential part of the `esmpy` conda package, moving the `make test_serial` block from `build.sh` to the `test:` block in the recipe would reduce the installed size of this package dramatically. Alternately if these tests should be run as part of the build process but the test datasets are not required for the final package, setting `$ESMPY_DATA_DIR` to some temporary directory that is not included in the package would suffice.

### Installed packages

```shell
# packages in environment at /home/hea211/hecking-around/esmpy-feedstock/.conda:
#
# Name                    Version                   Build  Channel
_libgcc_mutex             0.1                 conda_forge    conda-forge
_openmp_mutex             4.5                       2_gnu    conda-forge
beautifulsoup4            4.12.2             pyha770c72_0    conda-forge
blinker                   1.6.2              pyhd8ed1ab_0    conda-forge
boltons                   23.0.0             pyhd8ed1ab_0    conda-forge
boolean.py                4.0                pyhd8ed1ab_0    conda-forge
brotli-python             1.0.9           py311ha362b79_9    conda-forge
bzip2                     1.0.8                h7f98852_4    conda-forge
c-ares                    1.19.1               hd590300_0    conda-forge
ca-certificates           2023.7.22            hbcca054_0    conda-forge
certifi                   2023.7.22          pyhd8ed1ab_0    conda-forge
cffi                      1.15.1          py311h409f033_3    conda-forge
chardet                   5.2.0           py311h38be061_0    conda-forge
charset-normalizer        3.2.0              pyhd8ed1ab_0    conda-forge
click                     8.1.7           unix_pyh707e725_0    conda-forge
colorama                  0.4.6              pyhd8ed1ab_0    conda-forge
conda                     23.7.3          py311h38be061_0    conda-forge
conda-build               3.26.1          py311h38be061_0    conda-forge
conda-forge-pinning       2023.08.29.11.33.45      hd8ed1ab_0    conda-forge
conda-index               0.2.3              pyhd8ed1ab_0    conda-forge
conda-package-handling    2.2.0              pyh38be061_0    conda-forge
conda-package-streaming   0.9.0              pyhd8ed1ab_0    conda-forge
conda-smithy              3.24.1             pyhd8ed1ab_0    conda-forge
cryptography              41.0.3          py311h63ff55d_0    conda-forge
curl                      8.2.1                hca28451_0    conda-forge
deprecated                1.2.14             pyh1a96a4e_0    conda-forge
filelock                  3.12.3             pyhd8ed1ab_0    conda-forge
gettext                   0.21.1               h27087fc_0    conda-forge
git                       2.42.0          pl5321h86e50cf_0    conda-forge
gitdb                     4.0.10             pyhd8ed1ab_0    conda-forge
gitpython                 3.1.32             pyhd8ed1ab_0    conda-forge
glob2                     0.7                        py_0    conda-forge
gmp                       6.2.1                h58526e2_0    conda-forge
icu                       73.2                 h59595ed_0    conda-forge
idna                      3.4                pyhd8ed1ab_0    conda-forge
isodate                   0.6.1              pyhd8ed1ab_0    conda-forge
jinja2                    3.1.2              pyhd8ed1ab_1    conda-forge
jsonpatch                 1.32               pyhd8ed1ab_0    conda-forge
jsonpointer               2.0                        py_0    conda-forge
keyutils                  1.6.1                h166bdaf_0    conda-forge
krb5                      1.21.2               h659d440_0    conda-forge
ld_impl_linux-64          2.40                 h41732ed_0    conda-forge
libarchive                3.6.2                h039dbb9_1    conda-forge
libcurl                   8.2.1                hca28451_0    conda-forge
libedit                   3.1.20191231         he28a2e2_2    conda-forge
libev                     4.33                 h516909a_1    conda-forge
libexpat                  2.5.0                hcb278e6_1    conda-forge
libffi                    3.4.2                h7f98852_5    conda-forge
libgcc-ng                 13.1.0               he5830b7_0    conda-forge
libgomp                   13.1.0               he5830b7_0    conda-forge
libiconv                  1.17                 h166bdaf_0    conda-forge
liblief                   0.12.3               h27087fc_0    conda-forge
libnghttp2                1.52.0               h61bc06f_0    conda-forge
libnsl                    2.0.0                h7f98852_0    conda-forge
libsodium                 1.0.18               h36c2ea0_1    conda-forge
libsqlite                 3.43.0               h2797004_0    conda-forge
libssh2                   1.11.0               h0841786_0    conda-forge
libstdcxx-ng              13.1.0               hfd8a6a1_0    conda-forge
libuuid                   2.38.1               h0b41bf4_0    conda-forge
libxml2                   2.11.5               h232c23b_1    conda-forge
libzlib                   1.2.13               hd590300_5    conda-forge
license-expression        30.1.1             pyhd8ed1ab_0    conda-forge
lz4-c                     1.9.4                hcb278e6_0    conda-forge
lzo                       2.10              h516909a_1000    conda-forge
markupsafe                2.1.3           py311h459d7ec_0    conda-forge
more-itertools            10.1.0             pyhd8ed1ab_0    conda-forge
msrest                    0.6.21             pyh44b312d_0    conda-forge
ncurses                   6.4                  hcb278e6_0    conda-forge
oauthlib                  3.2.2              pyhd8ed1ab_0    conda-forge
openssl                   3.1.2                hd590300_0    conda-forge
packaging                 23.1               pyhd8ed1ab_0    conda-forge
patch                     2.7.6             h7f98852_1002    conda-forge
patchelf                  0.17.2               h58526e2_0    conda-forge
pcre2                     10.40                hc3806b6_0    conda-forge
perl                      5.32.1          4_hd590300_perl5    conda-forge
pip                       23.2.1             pyhd8ed1ab_0    conda-forge
pkginfo                   1.9.6              pyhd8ed1ab_0    conda-forge
pluggy                    1.3.0              pyhd8ed1ab_0    conda-forge
psutil                    5.9.5           py311h2582759_0    conda-forge
py-lief                   0.12.3          py311ha362b79_0    conda-forge
pycosat                   0.6.4           py311hd4cff14_1    conda-forge
pycparser                 2.21               pyhd8ed1ab_0    conda-forge
pycryptodome              3.18.0          py311h6601440_0    conda-forge
pygithub                  1.59.1             pyhd8ed1ab_0    conda-forge
pyjwt                     2.8.0              pyhd8ed1ab_0    conda-forge
pynacl                    1.5.0           py311hd4cff14_2    conda-forge
pyopenssl                 23.2.0             pyhd8ed1ab_1    conda-forge
pysocks                   1.7.1              pyha2e5f31_6    conda-forge
python                    3.11.5          hab00c5b_0_cpython    conda-forge
python-libarchive-c       5.0             py311h38be061_1    conda-forge
python_abi                3.11                    3_cp311    conda-forge
pytz                      2023.3             pyhd8ed1ab_0    conda-forge
pyyaml                    6.0.1           py311h459d7ec_0    conda-forge
readline                  8.2                  h8228510_1    conda-forge
requests                  2.31.0             pyhd8ed1ab_0    conda-forge
requests-oauthlib         1.3.1              pyhd8ed1ab_0    conda-forge
ripgrep                   13.0.0               h2f28480_2    conda-forge
ruamel.yaml               0.17.32         py311h459d7ec_0    conda-forge
ruamel.yaml.clib          0.2.7           py311h2582759_1    conda-forge
scrypt                    0.8.20          py311h6965a6d_0    conda-forge
setuptools                68.1.2             pyhd8ed1ab_0    conda-forge
six                       1.16.0             pyh6c4a22f_0    conda-forge
smmap                     3.0.5              pyh44b312d_0    conda-forge
soupsieve                 2.3.2.post1        pyhd8ed1ab_0    conda-forge
tk                        8.6.12               h27826a3_0    conda-forge
toolz                     0.12.0             pyhd8ed1ab_0    conda-forge
tqdm                      4.66.1             pyhd8ed1ab_0    conda-forge
typing_extensions         4.7.1              pyha770c72_0    conda-forge
tzdata                    2023c                h71feb2d_0    conda-forge
urllib3                   2.0.4              pyhd8ed1ab_0    conda-forge
vsts-python-api           0.1.25             pyhd8ed1ab_1    conda-forge
wheel                     0.41.2             pyhd8ed1ab_0    conda-forge
wrapt                     1.15.0          py311h2582759_0    conda-forge
xz                        5.2.6                h166bdaf_0    conda-forge
yaml                      0.2.5                h7f98852_2    conda-forge
zstandard                 0.19.0          py311haa97af0_2    conda-forge
zstd                      1.5.2                hfc55251_7    conda-forge
```

### Environment info

```shell
active environment : /home/hea211/hecking-around/esmpy-feedstock/.conda
    active env location : /home/hea211/hecking-around/esmpy-feedstock/.conda
            shell level : 1
       user config file : /home/hea211/.condarc
 populated config files : /home/hea211/.condarc
          conda version : 23.3.1
    conda-build version : 3.21.7
         python version : 3.9.16.final.0
       virtual packages : __archspec=1=x86_64
                          __glibc=2.31=0
                          __linux=5.15.0=0
                          __unix=0=0
       base environment : /home/hea211/.local/share/miniconda3  (writable)
      conda av data dir : /home/hea211/.local/share/miniconda3/etc/conda
  conda av metadata url : None
           channel URLs : https://conda.anaconda.org/conda-forge/linux-64
                          https://conda.anaconda.org/conda-forge/noarch
                          https://repo.anaconda.com/pkgs/main/linux-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/r/linux-64
                          https://repo.anaconda.com/pkgs/r/noarch
          package cache : /home/hea211/.local/share/miniconda3/pkgs
                          /home/hea211/.conda/pkgs
       envs directories : /home/hea211/.local/share/miniconda3/envs
                          /home/hea211/.conda/envs
               platform : linux-64
             user-agent : conda/23.3.1 requests/2.29.0 CPython/3.9.16 Linux/5.15.0-79-generic ubuntu/20.04.6 glibc/2.31
                UID:GID : 1214394:1214394
             netrc file : None
           offline mode : False
```

**Comment by xylar** [comment]
Okay I think I got this sorted out in #79.  I also sent a message to the ESMF team because I don't understand how `$ESMPY_DATA_DIR` was intended to work.  It seems like there's no way to download the data if you specify it.

**Comment by xylar** [comment]
Fixed in #79, see https://anaconda.org/conda-forge/esmpy/files.

Thanks @mx-moth!

**Comment by mx-moth** [comment]
Thanks for fixing this!
